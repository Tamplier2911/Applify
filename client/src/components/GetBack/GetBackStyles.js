import styled from "styled-components";

// svg
import { ReactComponent as ReturnSVG } from "../../assets/svg/return.svg";

export const GetBackContainer = styled.div`
  cursor: pointer;
  position: fixed;
  bottom: 4rem;
  left: 4rem;

  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 4rem;

  background-color: var(--cl-white);
  box-shadow: 0.1rem 0.2rem 0.6rem var(--cl-langpan-shadow);
  border-radius: 50%;
  width: 7rem;
  height: 7rem;

  transition: color 0.3s, transform 0.3s;

  &:hover {
    transform: scale(1.1);
  }

  &:hover * {
    fill: var(--cl-primary);
  }
`;

export const GetBackSVG = styled(ReturnSVG)`
  fill: var(--cl-font);
  width: 4rem;
  height: 4rem;

  transition: fill 0.3s;
`;
