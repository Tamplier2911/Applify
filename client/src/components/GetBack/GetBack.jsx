// import "./GetBack.scss";
import React from "react";
import { withRouter } from "react-router-dom";

// JS Rendering CSS
import { GetBackContainer, GetBackSVG } from "./GetBackStyles";

const GetBack = ({ history, path }) => {
  return (
    <GetBackContainer onClick={() => history.push(`${path}`)}>
      <GetBackSVG />
    </GetBackContainer>
  );
};

export default withRouter(GetBack);
