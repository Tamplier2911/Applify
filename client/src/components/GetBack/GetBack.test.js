import React from "react";
import toJson from "enzyme-to-json";

// component
import GetBack from "./GetBack";

// utils
import { setShallowR, findByTestAttr } from "../../tests/testUtils";

describe("<GetBack />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setShallowR(<GetBack />);
  });

  it("renders without an error", () => {
    const getBack = findByTestAttr(wrapper, "get-back");
    expect(getBack.length).toBe(1);
  });

  it("triggers onClick handler", () => {
    // define mock function
    let mockFn = jest.fn();
    // render initial component with router
    wrapper = setShallowR(<GetBack />);
    // rewrite history object with mock
    wrapper.setProps({ history: { push: mockFn } });
    // find desired object to initiate click
    const getBack = findByTestAttr(wrapper, "get-back");
    // initiate click
    getBack.simulate("click");
    // get calls length from mock
    const getClickCount = mockFn.mock.calls.length;
    // assert calls length to be 1
    expect(getClickCount).toBe(1);
  });

  it("renders children - SVG", () => {
    const getBackSvg = findByTestAttr(wrapper, "get-back-svg");
    expect(getBackSvg.length).toBe(1);
  });

  it("matches snapshot", () => {
    const getBack = findByTestAttr(wrapper, "get-back");
    expect(toJson(getBack)).toMatchSnapshot();
  });
});
