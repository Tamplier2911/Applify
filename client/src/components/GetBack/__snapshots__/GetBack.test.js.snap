// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GetBack /> matches snapshot 1`] = `
<styled.div
  data-test="get-back"
  onClick={[Function]}
>
  <Styled(Component)
    data-test="get-back-svg"
  />
</styled.div>
`;
